src/k8s/api/v1/cluster_config.go

```diff
@@ -13,7 +13,11 @@ type GetClusterConfigResponse struct {
 }
 
 type UpdateClusterConfigRequest struct {
-	Config UserFacingClusterConfig
+	Config UserFacingClusterConfig `json:"config,omitempty" yaml:"config,omitempty"`
+	// DatastoreConfig contains updates of the external datastore configuration.
+	// It is not part of the UserFacingClusterConfig as it is direcly marshalled by the CLI.
+	// The datastore config should not be displayed as part of the configuration in `k8s get`
+	DatastoreConfig UserFacingDatastoreConfig `json:"datastore-config,omitempty" yaml:"datastore-config,omitempty"`
```

Comment:
I would just call this `Datastore` and `json:"datastore"`. I would remove the comment.